### PR TITLE
feat(cmd/goat): allow specifying authFactorToken

### PR DESCRIPTION
**Pull request by lizthegrey**
Supports using goat account login with [email 2fa enabled](https://github.com/bluesky-social/atproto/discussions/2435) so that users do not need to first disable 2fa to then get an email token anyways to obtain plc signing tokens.

### CODE REVIEW

**Comment by bnewbold** [comment]
would close: https://github.com/bluesky-social/indigo/issues/787

**Review by bnewbold — APPROVED** [review]
generally looks good. left some style nits (i'm personally opinionated about this particular tool/codebase).

there is a potential annoying case where the session times out, so the tool tries to refresh using username/password, and it fails b/c a token is resolved. it would in theory be nice to have a more specific error message/code about that, but it is probably rare enough to not block merging this improvement for now.

thanks for the contribution! a few folks have hit this and I should have been less lazy about implementing it from the start.

File: cmd/goat/account.go [hunk 1]
```diff
@@ -163,7 +169,7 @@ func runAccountLogin(cctx *cli.Context) error {
 		return err
 	}
 
-	_, err = refreshAuthSession(ctx, *username, cctx.String("app-password"), cctx.String("pds-host"))
+	_, err = refreshAuthSession(ctx, *username, cctx.String("app-password"), cctx.String("auth-factor-token"), cctx.String("pds-host"))
```
[the remark this answers is not in the document]
Comment (lizthegrey) [reply cmd/goat/account.go]: Done.

File: cmd/goat/account.go [hunk 2]
```diff
@@ -37,6 +37,12 @@ var cmdAccount = &cli.Command{
 					Usage:    "password (app password recommended)",
 					EnvVars:  []string{"ATP_AUTH_PASSWORD"},
 				},
+				&cli.StringFlag{
+					Name:    "auth-factor-token",
+					Aliases: []string{"t"},
+					Usage:   "token required if password is used and 2fa is required",
+					EnvVars: []string{"ATP_AUTH_TOKEN"},
```
[the remark this answers is not in the document]
Comment (lizthegrey) [reply cmd/goat/account.go]: Done.  Although this makes it mismatch with the variable name in JSON schema which is AuthFactorToken. I can go with this or I can do ATP_AUTH_FACTOR_TOKEN. Up to you.

File: cmd/goat/account.go [hunk 3]
```diff
@@ -37,6 +37,12 @@ var cmdAccount = &cli.Command{
 					Usage:    "password (app password recommended)",
 					EnvVars:  []string{"ATP_AUTH_PASSWORD"},
 				},
+				&cli.StringFlag{
+					Name:    "auth-factor-token",
+					Aliases: []string{"t"},
```
[the remark this answers is not in the document]
Comment (lizthegrey) [reply cmd/goat/account.go]: Removed.

**Review by lizthegrey — COMMENTED** [review]
Thanks, all sorted.

File: cmd/goat/account.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lizthegrey) [reply cmd/goat/account.go]: did the latter thing, it feels right.

**Comment by bnewbold** [comment]
Thanks!

**Pull request merged**